web/src/chartOptions/dataZoom.ts

```diff
@@ -0,0 +1,36 @@
+import {WgsCovPlotDB} from "../db";
+import {isNil} from "lodash";
+
+export const getDataZoom: (db: WgsCovPlotDB)
+  => (any[] | [{ filterMode: string; xAxisIndex: number[]; type: string; startValue: number; endValue: number,  zoomLock: boolean }, { filterMode: string; xAxisIndex: number[]; showDataShadow: boolean; show: boolean; type: string; zoomLock: boolean }])
+  = (db: WgsCovPlotDB) => {
+  let xAxisIndex = [...Array(db.chartOptions.selectedSamples.length + 1).keys()];
+  let start: number = 1
+  let end: number = db.positions.length;
+  if (!isNil(db.chart)) {
+    if (!isNil(db.chart.getOption())) {
```

Comment:
@nhhaidee Instead of calling the same function twice, can you refactor to only call it once and save the result to a variable?

```js
let opts = db.chart.getOption();
if (!isNil(opts)) {
  let dataZoom: any = opts.dataZoom;
...
}
```

Reply:
I tried with `!isNil(db.chart.getOption())` only, but at the beginning when the chart has not yet created, it will throw error `Uncaught TypeError: Cannot read properties of null (reading 'getOption')`. Please let me know if you have better solution